### ISSUE TITLE
Invalid HTTP Status

### ISSUE THREAD

**Issue by jammerxd** [socketio/socket.io-client-cpp#150]
Websocketpp is failing with Invalid HTTP Status...

**Comment by FinnGu** [comment]
Did you resolve this issue? I am getting the same error for some days now...

`[2017-10-04 18:46:23] [connect] Successful connection
[2017-10-04 18:46:23] [error] Server handshake response error: websocketpp.processor:20 (Invalid HTTP status.)
Connection failed.
Reconnect for attempt:4`

**Comment by jammerxd** [comment]
Yes I did, I had to make some changes to allow for https connections.

**Comment by FinnGu** [comment]
I ended up resolving this issue by using a different underlying service for my server socketio implementation:
Instead of using **gevent** which relies on long-polling, I now use **eventlet** which additionally supports websocket transport and allows for communication with this client out of the box.

**Comment by aolivier** [comment]
@jammerxd care to share your changes?

**Comment by jammerxd** [comment]
Sure, give me a day or 2.

**Comment by jammerxd** [comment]
@aolivier I uploaded my mods to this repo: https://github.com/jammerxd/secure-socket.io-client-cpp

**Comment by austinpaine** [comment]
Soooo is this still open?  Currently seeing this same error in my application and cant seem to get past it. Relevant stack trace:
``****

**Comment by johann8384** [comment]
Me as well, looks like it is still an issue:

```Jun 13 16:45:24 medrc-recorder-1 recorder[19546]: [2018-06-13 16:45:24] [connect] Successful connection
Jun 13 16:45:24 medrc-recorder-1 recorder[19546]: [2018-06-13 16:45:24] [error] Server handshake response error: websocketpp.processor:20 (Invalid HTTP status.)
Jun 13 16:45:24 medrc-recorder-1 recorder[19546]: [2018-06-13 16:45:24] [application] on_fail: WebSocket Connection failed, stopping telemetry!
Jun 13 16:45:24 medrc-recorder-1 recorder[19546]: [2018-06-13 16:45:24] [application] on_fail: Will try to reconnect in:  19

**Comment by ImSanjayChintha** [comment]
Did any one got how to resolve this issue?

**Comment by polishchlieb** [comment]
Maybe just try the `tls` branch? I'll check it in a day.

**Comment by gituser9** [comment]
Any news on the issue?

**Comment by alter-sachin** [comment]
Any updates ?

**Comment by AAbhishek1234** [comment]
Hi guys , Any updates on this ?

**Comment by jammerxd** [comment]
I thought I found the issue was because I was trying to go with wss;//<ip or domain> instead of https://<ip or domain> and having the web socket server be behind nginx and having nginx upgrade the connection...I can't remember if that was the issue or not.

**Comment by AAbhishek1234** [comment]
for me issue was the tls version was wrong as this library is not upto date + there was no way to add path . I change the codebase and it worked . cheers

**Comment by rthickling** [comment]
I still have this problem.
What was the change that fixed it?  Any links?

**Comment by wangbing8895** [comment]
@jammerxd I also meet this issue.when I connect to a wss server without nginx,it works;but when I connect a wss server with nginx,it doesn't work! why?

**Comment by zak905** [comment]
any updates about this?

**Comment by Laateef** [comment]
I encountered the same problem and resolved it by appending "/" to the resource path:
> **https://somehost/somepath** -> **https://somehost/somepath/**

**Comment by zeze-zeze** [comment]
Using vcpkg, just install openssl-windows, then compile the project with tls version sioclient.lib.
If you use visual studio, set the properties => vcpkg => Use Static Libraries to yes. libcrypto.dll and libssl.dll will be built when the project is compiled.

But now I am wondering why we set the property to no, tls does not work properly.

**Comment by asdkmm5050** [comment]
@zeze-zeze Hi, any update ?
